### PR TITLE
Feature/s3 c 3755 add bucket delete encryption handler

### PR DESCRIPTION
**Pull request by tmacro**
~Note for reviewers: Arsenal has been bumped to the hash of the Arsenal PR branch, and will be changed after it has merged.~

### CODE REVIEW

File: lib/api/apiUtils/bucket/bucketEncryption.js [hunk 1]
```diff
@@ -108,7 +109,31 @@ function parseBucketEncryptionHeaders(headers) {
...
+function checkExpectedBucketOwner(headers, bucket, log, cb) {
+    const expectedOwner = headers['x-amz-expected-bucket-owner'];
+    if (expectedOwner === undefined) {
+        return cb();
+    }
+
+    const bucketOwner = bucket.getOwner();
+    return vault.getAccountIds([bucketOwner], log, (error, res) => {
+        if (error) {
+            log.error('error fetch accountId from vault', {
```
Comment (jonathan-gramain) [inline lib/api/apiUtils/bucket/bucketEncryption.js:121]:
```suggestion
            log.error('error fetching accountId from vault', {
```

File: lib/api/apiUtils/bucket/bucketEncryption.js [hunk 2]
```diff
@@ -108,7 +109,31 @@ function parseBucketEncryptionHeaders(headers) {
     };
 }
 
+function checkExpectedBucketOwner(headers, bucket, log, cb) {
```
Comment (jonathan-gramain) [inline lib/api/apiUtils/bucket/bucketEncryption.js:112]: Why do we check this header specifically for delete-bucket-encryption? According to https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucket-owner-condition.html, it looks to me this header is generic to pretty much all object and bucket operations, and optional, so kinda looks orthogonal to bucket encryption.

File: tests/unit/api/bucketDeleteEncryption.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                return getSSEConfig(bucketName, (err, sseInfo) => {
+                    assert.ifError(err);
+                    assert.strictEqual(sseInfo.mandatory, false);
+                    done();
+                });
+            });
+        });
+    });
+
+    it('should disable mandatory sse for aws:kms', done => {
```
Comment (jonathan-gramain) [inline tests/unit/api/bucketDeleteEncryption.js]: You might want to merge this test with the preceeding one and a `forEach` on the algorithm since the logic is essentially the same

File: lib/api/apiUtils/bucket/bucketEncryption.js (hunk 2 above)
Comment (tmacro) [reply lib/api/apiUtils/bucket/bucketEncryption.js:112]: It was accidentally skipped in put/get. I've created a ticket to add support for them.

Comment (jonathan-gramain) [reply lib/api/apiUtils/bucket/bucketEncryption.js:112]: OK sounds good

Ideally I think it should not be defined in a bucket-encryption-specific file but it should be part of the common checks path for all request types.

**Review by jonathan-gramain — APPROVED** [review]

File: lib/api/apiUtils/bucket/bucketEncryption.js (hunk 2 above)
Comment (tmacro) [reply lib/api/apiUtils/bucket/bucketEncryption.js:112]: Added a note in the ticket to look at moving it.

**Review by naren-rajendran — APPROVED** [review]
